gwdetchar/lasso/tests/test_main.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for gwdetchar.lasso.__main__
+"""
+
+import tempfile
+
+import numpy as np
+
+from gwpy.timeseries import TimeSeries
+
+from .. import __main__ as lasso
+
+
+__author__ = 'Michael Lowry <michaeljohn.lowry@ligo.org>'
+
+
+# global test objects
+
+# volatile temp file to write expected timeseries to
+TEMP = tempfile.NamedTemporaryFile()
+TEST_CHANNEL = 'L1:DMT-SNSW_EFFECTIVE_RANGE_MPC.mean'
+TEST_START = 0
+TEST_END = TEST_START+1000
+TEST_ARRAY = np.linspace(TEST_START, TEST_END, num=TEST_END-TEST_START)
+expected_ts = TimeSeries(TEST_ARRAY, t0=TEST_START, dt=1, channel=TEST_CHANNEL)
+expected_ts.write(TEMP, format='gwf')
```

Comment:
This should be moved into a fixture function to be evaluated only when the test that requires it is called:

```suggestion
TEST_CHANNEL = 'L1:DMT-SNSW_EFFECTIVE_RANGE_MPC.mean'
TEST_START = 0
TEST_END = TEST_START+1000


@pytest.fixture
def primary_ts_file(tmp_path):
    # construct random data to read
    data = np.linspace(TEST_START, TEST_END, num=TEST_END-TEST_START)
    ts = TimeSeries(data, t0=TEST_START, dt=1, channel=TEST_CHANNEL)
    
    # write data to a file and return that
    outfile = tmp_path / "data.gwf"
    ts.write(outfile, format='gwf')
    return outfile
```

Reply:
This looks like it has been resolved following Duncan's suggestion